discovery/discovery.go

```diff
@@ -63,6 +64,7 @@ func main() {
 		rotator: r,
 	}
 	d.cond = sync.NewCond(&d.mtx)
+	go d.keepAliver()
```

Comment:
How is this goroutine exited when a client disconnects?

Reply:
It doesn't, there's a single thread for the entire discovery server. (Note that they all share the same lock+convar)